src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompilerExtensionTargetFrameworkAnalyzer.cs

```diff
@@ -0,0 +1,136 @@
+                    return;
+
+                AttributeData? appliedTargetFrameworkAttribute = context.Compilation.Assembly.GetAttribute(targetFrameworkAttribute);
+                if (appliedTargetFrameworkAttribute is null)
+                    return;
+
+                if (appliedTargetFrameworkAttribute.ConstructorArguments.IsEmpty)
+                {
+                    return;
+                }
+
+                string displayName;
+                switch (appliedTargetFrameworkAttribute.ConstructorArguments[0].Value as string)
+                {
+                    case ".NETStandard,Version=v1.0":
+                    case ".NETStandard,Version=v1.1":
+                    case ".NETStandard,Version=v1.2":
+                    case ".NETStandard,Version=v1.3":
+                    case ".NETStandard,Version=v1.4":
+                    case ".NETStandard,Version=v1.5":
+                    case ".NETStandard,Version=v1.6":
+                    case ".NETStandard,Version=v2.0":
+                        // The compiler supports this target framework
+                        return;
+
+                    default:
+                        displayName = appliedTargetFrameworkAttribute.NamedArguments.FirstOrDefault(arg => arg.Key == nameof(TargetFrameworkAttribute.FrameworkDisplayName)).Value.Value as string
+                            ?? appliedTargetFrameworkAttribute.ConstructorArguments[0].Value as string
+                            ?? "<unknown>";
+                        break;
+                }
+
+                var diagnosticAnalyzerAttribute = typeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisDiagnosticsDiagnosticAnalyzerAttribute);
+                var sourceGeneratorInterface = typeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisISourceGenerator);
+                var incrementalGeneratorInterface = typeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisIIncrementalGenerator);
+                var generatorAttribute = typeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute);
+
+                context.RegisterSymbolAction(
+                    context =>
+                    {
+                        var namedType = (INamedTypeSymbol)context.Symbol;
+                        if (!IsRegisteredExtension(namedType, diagnosticAnalyzer, diagnosticAnalyzerAttribute, out var applicationSyntaxReference)
+                            && !IsRegisteredExtension(namedType, sourceGeneratorInterface, generatorAttribute, out applicationSyntaxReference)
+                            && !IsRegisteredExtension(namedType, incrementalGeneratorInterface, generatorAttribute, out applicationSyntaxReference))
+                        {
+                            // This is not a compiler extension
+                            return;
+                        }
```

Comment:
Is it intentional to not check `CodeFixProvider`?

Reply:
➡️ Yes. CodeFixProvider is an IDE feature, not a compiler feature.